app/models.js

```diff
@@ -45,27 +45,44 @@ const fetchCommentsById = (article_id) => {
 };
 
 const addComment = (comment, article_id) => {
-    return db.query(`SELECT * FROM articles WHERE article_id = $1`, [article_id])
-        .then(({ rows, rowCount }) => {
-            if (rowCount === 0) {
-                return Promise.reject({ status: 404, msg: "Not found - no article of this ID in database" })
-            }
-            else {
-                const correctArticle = rows[0]
-                const formattedComment = [[comment.body, article_id, correctArticle.author]]
+    return fetchArticleById(article_id)
+        .then(() => {
+            const formattedComment = [[comment.body, article_id, comment.username]]
 
-                let sqlAddCommentString = format(`INSERT INTO comments
+            let sqlAddCommentString = format(`INSERT INTO comments
                                     (body, article_id, author)
                                     VALUES
                                     %L
                                     RETURNING *`, formattedComment)
 
-                return db.query(sqlAddCommentString)
-                    .then(({ rows }) => {
-                        return rows[0];
-                    });
+            return db.query(sqlAddCommentString)
+        })
+        .then(({ rows }) => {
+            return rows[0];
+        });
+};
+
+const updateVotes = (body, article_id) => {
+    return fetchArticleById(article_id)
+        .then(() => {
+            return db.query(`SELECT votes FROM articles WHERE article_id = $1`, [article_id])
+        })
+        .then(({ rows }) => {
+            if (!body.inc_votes || typeof body.inc_votes !== "number") {
+                return Promise.reject({ status: 400, msg: "Bad request - body must have 'inc_votes' property of 'number' data type" })
             }
+            let newVoteTotal = rows[0].votes + body.inc_votes;
```

Comment:
There is a much simpler way to update the votes without making a query to get the votes first (a model should ideally only make one query to the database). It is possible to add the increase amount value to current value using SQL. Your current sqlUpdateVotesQuery string is very close!

If you get this correct then you won't need the other query above 😄

Reply:
Figured it out - thank you!